pkg/kinflate/commands/configmap.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package commands
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+)
+
+type AddConfigMap struct {
+	// Name of configMap (required)
+	Name string
+	// FileSources to derive the configMap from (optional)
+	FileSources []string
+	// LiteralSources to derive the configMap from (optional)
+	LiteralSources []string
+	// EnvFileSource to derive the configMap from (optional)
+	EnvFileSource string
+}
+
+// validate validates required fields are set to support structured generation.
+func (a *AddConfigMap) Validate() error {
+	if len(a.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+	if len(a.EnvFileSource) > 0 && (len(a.FileSources) > 0 || len(a.LiteralSources) > 0) {
+		return fmt.Errorf("from-env-file cannot be combined with from-file or from-literal")
+	}
+	return nil
+}
+
+func NewCmdAddConfigMap(errOut io.Writer) *cobra.Command {
+	var config AddConfigMap
+	cmd := &cobra.Command{
+		Use:     "configmap NAME [--from-file=[key=]source] [--from-literal=key1=value1]",
+		Short:   "Adds a configmap to your manifest file.",
+		Long:    "",
+		Example: "",
+		RunE: func(_ *cobra.Command, args []string) error {
+			err := config.Validate()
+			if err != nil {
+				return err
+			}
+
+			if len(args) != 1 {
+				return fmt.Errorf("error: exactly one NAME is required, got %d", len(args))
+			}
+			config.Name = args[0]
```

Comment:
suggest Validate accepts args in line 56 so you have something to validate.

also, please add test

Reply:
Yeah, we'll add tests later